src/formatter.rs

```diff
@@ -31,6 +33,7 @@ pub(crate) fn format(input: impl Read, mut output: impl Write) -> Result<(), io:
     loop {
         buf.clear();
         input.read_until(una.segment_delimiter, &mut buf)?;
+        debug!(segment = ?String::from_utf8_lossy(&buf), "Formatting segment");
```

Comment:
small note: You don't need to use `?` when the value itself implements [`tracing::Value`](https://docs.rs/tracing/latest/tracing/trait.Value.html), 

and if you don't need the debug representation, you can use `%` to just get the `Display` value. 

In this specific case I would guess you don't need any `%` or `?`, and if, `%` is more fitting

Reply:
I see, good to know. I see that `from_utf8_lossy` return `Cow<>` which seems like not implementing `tracing::Value`
But display is there. I will use `%`. 

Error I see when I don't prepend anything:
```
rustc: the trait bound `std::borrow::Cow<'_, str>: tracing::Value` is not satisfied
the following other types implement trait `tracing::Value`:
  &'a T
  &'a mut T
  (dyn std::error::Error + 'static)
  (dyn std::error::Error + std::marker::Send + 'static)
  (dyn std::error::Error + std::marker::Send + std::marker::Sync + 'static)
  (dyn std::error::Error + std::marker::Sync + 'static)
  bool
  f32
and 34 others
required for the cast from `&std::borrow::Cow<'_, str>` to `&dyn tracing::Value` [E0277]

```